### PR TITLE
Add File/Image message handling

### PR DESCRIPTION
**Pull request by raccube**
Provides chat links for messages with a file or image attachment to prevent them from being sent as "File" or "Image" in-game.

### CODE REVIEW

File: src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt [hunk 1]
```diff
@@ -13,7 +20,33 @@ class MatrixRoomMessageReceivedListener(private val plugin : Main) : RoomMessage
...
+            val size = FileUtils.byteCountToDisplaySize(messageContent.size!!.toLong())
+            sendLinkMessage(plugin, url, "File [${messageContent.filename}, $size]")
+            return true
+        }
+
         plugin.server.broadcastMessage("[§2${event.msg.sender}§r] ${event.msg.body}")
         return true
     }
+
+    private fun sendLinkMessage(plugin: Main, url: String, label: String) {
```
Comment (nilsding) [inline src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt]: There is no need for passing `plugin` to this method -- it's already available as an instance variable.  Please remove it from the both method signatures.

**Review by nilsding — APPROVED** [review]

File: src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt (hunk 1 above)
Comment (nilsding) [reply src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt]: Oida

**Review by nilsding — CHANGES_REQUESTED** [review]

**Comment by raccube** [comment]
Forgot to actually add who sent the message.

File: src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt [hunk 2]
```diff
@@ -13,7 +18,61 @@ class MatrixRoomMessageReceivedListener(private val plugin : Main) : RoomMessage
...
+    }
+
+    private fun sendMessage(message: Array<BaseComponent>) {
+        plugin.server.onlinePlayers.forEach { it.spigot().sendMessage(ChatMessageType.CHAT, *message) }
+    }
+
+    private fun createMessagePrefix(message: Message): Array<BaseComponent> {
+        val sender = TextComponent(message.sender.toString())
+        sender.color = ChatColor.DARK_GREEN
+        val tooltip = ComponentBuilder(message.sender.displayname)
```
Comment (nilsding) [inline src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt]: Please avoid using the `sender.displayname` as it makes a request to matrix.org, and that delays the forwarding of the message quite a bit

File: src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt [hunk 3]
```diff
@@ -13,7 +18,61 @@ class MatrixRoomMessageReceivedListener(private val plugin : Main) : RoomMessage
             return true
         }
 
-        plugin.server.broadcastMessage("[§2${event.msg.sender}§r] ${event.msg.body}")
+        val content = event.msg.content
+        val message = createMessagePrefix(event.msg)
+        val description = getMediaDescription(content)
+        if (content.msgtype == MessageTypes.TEXT) {
+            message[3] = TextComponent(content.body)
```
Comment (nilsding) [inline src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt]: This does not look safe

**Review by nilsding — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt [hunk 4]
```diff
@@ -13,7 +18,67 @@ class MatrixRoomMessageReceivedListener(private val plugin : Main) : RoomMessage
...
+
+        value.add(TextComponent("["))
+        value.add(sender)
+        value.add(TextComponent("] "))
+
+        return value
+    }
+
+    private fun getMediaUrl(message: UrlMessageContent) =
+        "${plugin.matrixClient!!.hs.base}media/v1/download${message.url!!.mediaId}"
```
Comment (nilsding) [inline src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt]: Add a `/`

**Review by nilsding — CHANGES_REQUESTED** [review]

File: src/main/resources/config.yml [hunk 5]
```diff
@@ -9,3 +9,6 @@ password: "changeme"
 
 # The room ID forward send messages from/to
 room_id: "!foo:example.com"
+
+# Adds a tooltip to the sender
+sender_tooltips: false
```
Comment (nilsding) [inline src/main/resources/config.yml]: also add a trailing newline

**Review by nilsding — CHANGES_REQUESTED** [review]

**Review by nilsding — APPROVED** [review]

**Pull request merged**